File: 6주차/이연걸/이중우선순위큐.py
from heapq import heappush, heappop
def solution(operations):
    heap = []
    
    for op in operations:
        if op[0] == 'I':
            heappush(heap, int(op[2:]))
        elif len(heap) > 0 and op == "D -1":
            heappop(heap)
        elif len(heap) > 0 and op == "D 1":
            heap.pop()
    
    if len(heap) == 0:
        return [0, 0]
    return [max(heap), min(heap)]

File: 6주차/임유정/이중우선순위큐.js
// https://school.programmers.co.kr/learn/courses/30/lessons/42628

function solution(operations) {
  var answer = [];
  const list = [];
  operations.forEach((o) => {
    const type = o[0];
    const value = Number(o.split(" ")[1]);

    if (type === "I") {
      list.push(value);
      list.sort((a, b) => a - b);
    } else if (type === "D" && list.length) {
      if (value === -1) {
        list.shift();
      } else if (value === 1) {
        list.pop();
      }
    }
  });
  answer = list.length ? [Math.max(...list), Math.min(...list)] : [0, 0];
  return answer;
}
